### PR TITLE
[DEV 4347] Added support for filter in preproc

### PR DESCRIPTION
**Pull request by nonibansal**

### CODE REVIEW

**Comment by aditya-nambiar** [comment]
We have a function called lambda_to_python_regular_func that helps convert lambda to regular func and is used in filter. Can we use that ?

**Comment by aditya-nambiar** [comment]
lambda's tend to break when they are multi line/nested lambda etc, can you add some test cases around them ? if you use the above lib function I'm hopeful it would work out

**Review by aditya-nambiar — CHANGES_REQUESTED** [review]

File: docs/examples/concepts/source.py [hunk 1]
```diff
@@ -69,3 +69,29 @@ class Order:
...
+def test_filter_preproc_source(client):
+    s3 = S3(name="mys3")
+
+    # docsnip filter
+    # docsnip-highlight start
+    @source(
+        s3.bucket("data", path="*"),
+        disorder="1w",
+        cdc="append",
+        filter=lambda x: x["skuid"] >= 1000,
```
[the remark this answers is not in the document]
Comment (nonibansal) [reply docs/examples/concepts/source.py:84]: renamed

**Review by nikhilgarg28 — APPROVED** [review]
Didn't review carefully since @aditya-nambiar had done that. 
That said, want to make sure the following are done:
1. Multiline lambdas work
2. If a user passes a function (not lambda) along with includes, that works
3. Preproc works on mock client too.

If any of these don't work, could you plz address them in a follow up commit.

**Comment by nonibansal** [comment]
> Didn't review carefully since @aditya-nambiar had done that. That said, want to make sure the following are done:
> 
> 1. Multiline lambdas work
> 2. If a user passes a function (not lambda) along with includes, that works
> 3. Preproc works on mock client too.
> 
> If any of these don't work, could you plz address them in a follow up commit.

@aditya-nambiar Made the change to allow multi line lambda.
Function is also allowed in filter along with lambda.
Preproc filter works on mock client too.

**Pull request merged**